File: 4131-88-1493-longest-subarray-of-1s-after-deleting-one-element/4131-88-1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int l=0,count=0,num=0;

        for(int i=0;i<nums.length;i++){
            if(nums[i]==0)
                num++;
            if(num>1){
                while(nums[l]==1)
                    l++;
                l++;
                num--;
            }
            count = Math.max(count,i-l);
        }
        return count;
    }
}

File: 4131-88-1493-longest-subarray-of-1s-after-deleting-one-element/4131-88-1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l,count,num = 0,0,0

        for i in range(len(nums)):
            if nums[i]==0:
                num+=1
            if num>1:
                while nums[l]==1:
                    l+=1
                l+=1
                num-=1
            count = max(count,i-l)
        return count